src/replication.c

```diff
@@ -2609,6 +3225,16 @@ int slaveTryPartialResynchronization(connection *conn, int read_reply) {
         return PSYNC_TRY_LATER;
     }
 
+    if (!strncmp(reply, "-FULLSYNCNEEDED", 15)) {
+        /* A response of -FULLSYNCNEEDED from the master implies that partial 
+         * synchronization is not possible. Full sync will continue on dedicated 
+         * RDB channel. */
```

Comment:
Mention that this is also how we find out that the primary supports this.

```suggestion
        /* A response of -FULLSYNCNEEDED from the master implies that partial 
         * synchronization is not possible and that the primary supports full
         * sync using dedicated RDB channel. Full sync will continue that way. */
```